src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -363,7 +370,7 @@ protected function instantiateObject(array &$data, string $class, array &$contex
 
                         $variadicParameters = [];
                         foreach ($data[$paramName] as $parameterData) {
-                            $variadicParameters[] = $this->denormalizeParameter($reflectionClass, $constructorParameter, $paramName, $parameterData, $context, $format);
+                            $variadicParameters[] = $this->denormalizeParameter($reflectionClass, $constructorParameter, $paramName, $parameterData, $context + $attributeContext, $format);
```

Comment:
attribute context is more specific, so it should override keys from the `$context`: use `array_merge` instead ?

```suggestion
                            $variadicParameters[] = $this->denormalizeParameter($reflectionClass, $constructorParameter, $paramName, $parameterData, array_merge($context, $attributeContext), $format);
```

Reply:
Well, sure. If this is the approach taken normally i can adjust it. Didnt find it in the codebase and went for the "save" route :D

Reply:
That's the approach introduced in the original feature yes: https://github.com/symfony/symfony/blob/7d4b42cbeef195e0a01272b9c5f464f0afe52542/src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php#L228-L252